File: os/escalonador01.c
#include <dirent.h>
#include <stdio.h>
#include <string.h>

int main() 
{
    DIR *d;

    struct dirent *dir;    

    d = opendir("processos");

    if (d)
    {
        while ((dir = readdir(d)) != NULL)
        {
            if (strcmp(dir->d_name,".") != 0 && strcmp(dir->d_name,"..") != 0)
            	printf("Processo: %s\n", dir->d_name);
        }
        closedir(d);
    }
    return(0);
}

File: os/escalonador01.php
<?php
define('SOURCE',__DIR__ . "/processos");
$files = scandir(SOURCE);
unset($files[0]);
unset($files[1]);
foreach($files as $file){
	echo "Executando processo $file...\n";
        $handle = fopen(SOURCE . "/$file",'r');
        while (!feof($handle)){
            $instruction = fgets($handle,6);
            if (!empty(trim($instruction))) {
	            echo "Executando instrução " . $instruction . "\n";
	    }
        }
        fclose($handle);
}
